kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -135,3 +93,56 @@ func enableComputeAPI(project string) error {
 
 	return nil
 }
+
+// getProjectFromBoskos creates a boskos client, acquires a gcp project
+// and starts a heartbeat goroutine to keep the project reserved
+func (d *deployer) getProjectFromBoskos() error {
+	boskos, err := client.NewClient(
+		os.Getenv("JOB_NAME")+"-kubetest2",
+		d.BoskosLocation,
+		"",
+		"",
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create boskos client: %s", err)
+	}
+
+	resourceType := "gce-project"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
```

Comment:
generally, I've seen context should be passed in and acquired from background at the top level. This lets you share auth and other metadata easily. Gj on adding a timeout here though.

Reply:
You'll probably want this boskos timeout to be configurable at the top level.  Or at the very least, a const at the top of the file.

Reply:
Made the timeout configurable. I'm not sure about grabbing a context at the top level, given that this is currently the only place it is needed and I don't (currently) see a use case for sharing auth/other metadata outside of the deployer struct itself.

Reply:
That sgtm, we can revisit if there are more requests being made from the deployer.